pkg/reconciler/apibinding/name_conflict_checker.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apibinding
+
+import (
+	"fmt"
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+	apislisters "github.com/kcp-dev/kcp/pkg/client/listers/apis/v1alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	crdlister "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/clusters"
+)
+
+type nameConflictChecker struct {
+	apiBindingLister        apislisters.APIBindingLister
```

Comment:
(No action required) Suggest providing function signatures (e.g. listAPIBinding, getAPIExport, etc) so that they can be trivially substituted in unit tests.

Reply:
Am working on this 😄